R/plink_calls.R

```diff
@@ -66,12 +69,22 @@ p_2_b <- function(ped_file, bed_file=ped_file, del=TRUE, plink_path=TRUE) {
 #' @export
 analysis_association <- function(geno_file, pheno_file, pheno_name, out_file,
                                  bed=TRUE, plink_path=TRUE) {
-  stopifnot("geno_file needs to be a valid file" = file.exists(geno_file),
-            "pheno_file needs to be a valid file" = file.exists(pheno_file),
-            "pheno_name" = TRUE,  # fix det Rasmus
-            "out_file" = TRUE,  # fix det Rasmus
-            "bed needs to be either TRUE or FALSE" = class(bed) == "logical",
-            "plink_path needs to be a valid path to plink" = (plink_path == TRUE || file.exists(paste0(plink_path, "/plink.exe"))))
+  stopifnot("geno_file needs to be a valid file" =
+              (file.exists(geno_file) &&
+                 (tools::file_ext(geno_file) == "ped" ||
+                    tools::file_ext(geno_file) == "bed")),
+            "pheno_file needs to be a valid file" =
+              (file.exists(pheno_file) &&
+                 (tools::file_ext(pheno_file) == "bed" ||
+                    tools::file_ext(output_file) == "ped")),
```

Comment:
pheno_file doesn't use bed or ped extensions, it is a .txt file

Reply:
i have fixed this for the four files i looked at, the only reason i didn't fix it for the rest is to avoid conflicts